docs/guide/custom-login-requirements.md

```diff
@@ -89,14 +90,15 @@ To illustrate, the general structure on the server side is:
 
 Each distinct requirement also has browser-side code, which is implemented as a Vue component. As explained in the [custom UI guide](custom-ui.md), Vue components intended for the admin UI (including login requirements) must be placed in the `ui/apos/components` subdirectory of a module in the project. For this purpose they are typically placed in the `modules/@apostrophecms/login/ui/apos/components` module at project level, or in an npm module that enhances `@apostrophecms/login` via `improve`.
 
-The developer is responsible for the appearance of the component. For `beforeSubmit` requirements, the component will appear at the bottom of the login form itself. For `afterPasswordVerified` requirements, it will appear on its own, after the Login button is clicked and the password is verified. If there are multiple `afterPasswordVerified` requirements, the user will see them presented one at a time. Transitions are provided by Apostrophe and do not need to be included in login requirement components.
+The developer is responsible for the appearance of the component. For `beforeSubmit` and `uponSubmit` requirements, the component will appear at the bottom of the login form itself. For `afterPasswordVerified` requirements, it will appear on its own, after the Login button is clicked and the password is verified. If there are multiple `afterPasswordVerified` requirements, the user will see them presented one at a time. Transitions are provided by Apostrophe and do not need to be included in login requirement components.
 
 Each component is responsible for:
 
-* Presenting its own UI, if any. Note that `afterPasswordVerified` requirements will appear in isolation, one at a time, while any UI for `beforeSubmit` components will appear simultaneously with the login form.
+* Presenting its own UI, if any. Note that `afterPasswordVerified` requirements will appear in isolation, one at a time, while any UI for `beforeSubmit` and `uponSubmit` components will appear simultaneously with the login form.
 * Accepting the properties of the object returned by the server-side `props(req)` function as props.
 * Emitting a `done` event with a payload providing proof that the requirement has been met. This proof is accessible to the `verify(req)` function on the server side as `req.body.requirements.RequirementName`. In the case of `afterPasswordVerified` requirements, the `done` event should not be emitted until the user has indicated their response is complete in some way.
 * `beforeSubmit` requirements may emit a `block` event to cancel a previous `done` event so that the Login button cannot be clicked yet.
+* `uponSubmit` requirements may emit a `block` event to cancel a previous `done` event so that the Login can be clicked again.
```

Comment:
Login -> Login button

Reply:
Line 97 isn't quite right. "any UI for `beforeSubmit` will appear simultaneously with the login form. Any UI for `uponSubmit` will appear after the Login button is clicked and remain until each has emitted a `done` event.